frame/alliance/src/migration.rs

```diff
@@ -41,10 +42,10 @@ pub fn migrate<T: Config<I>, I: 'static>() -> Weight {
 	weight
 }
 
-/// Implements `OnRuntimeUpgrade` trait.
+/// Implements `on_runtime_upgrade()` from Hooks trait.
 pub struct Migration<T, I = ()>(PhantomData<(T, I)>);
 
-impl<T: Config<I>, I: 'static> OnRuntimeUpgrade for Migration<T, I> {
+impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Migration<T, I> {
```

Comment:
Why are we removing `OnRuntimeUpgrade` @kianenigma ?  
I get the `OnInitialize` and `OnFinlize`, but `OnRuntimeUpgrade` is not really related to pallets.

Reply:
Maybe `Hooks` should have never copied all the trait methods. Instead it should have been a trait that collects all the important traits as super traits.

Reply:
Yeah I think @bkchr's suggestion makes the right sense here, `Hooks` should be a mere supertrait, and we can then keep the old traits for when a fine-grained specification is needed, such as the case of `OnRuntimeUpgrade`.

Reply:
Okay, so now this PR does not make sense, instead a PR is require that makes Hooks a super trait.

Reply:
@kianenigma @bkchr could you please explain following:
if we make Hooks a super trait then what is the point of having it?
AFAIU all pallets needs to implement old traits and Hooks does not add anything on top of what individual traits like `OnInitialize` ... adds.
Kindly correct me if my understanding is wrong here.

Reply:
It wouldn't add any functionality in itself, and it would just be a marker trait. Its main purpose would be that inside the pallet macros you would only implement `Hooks` and not 5 or 6 different traits.

Reply:
In the past the pallet macro was adding a bit more than just logging when implementing `OnRuntimeUpgrade` trait.

So it felt a bit abusive to add code in the implementation of the `Hooks::on_runtime_upgrade` function block.
The idea was to add as less magic as possible inside the code that is written by the user. Generally pallet macro don't modify their code at all just use it to implement more traits and types.

Now that the only added logic is logging and span it can be ok.
But still when user call this `Hooks::on_runtime_upgrade` function in the tests or anywhere it could be confusing to see more logging and span than expected.

Also it means that in the future if more code needs to be added by the pallet macro then what the user see and what the implementation do will differ again, which is a bit confusing.